### ISSUE TITLE
multi-occur and helm-mode

### ISSUE THREAD

**Issue by amelio-vazquez-reina** [emacs-helm/helm#460]
One of my favorite ways of searching inside buffers is to:
1. `M-x helm-buffers-list`
2. Mark buffers with `C-<SPACE>` (or `M-a`after narrowing the list a bit) 
3. `C-s`
4. Start typing text to search, and Helm will show a list.

I am not sure if the above uses `helm-multi-occur` behind the scenes, so I decided to try  `multi-occur` manually, but when I do so, my commands  enter an infinite loop. 

I read on the mailing list the following thread "[helm multi-occur infinite loop](http://permalink.gmane.org/gmane.emacs.helm.user/197)". The answer seems to be that I am entering an infinite loop because I  have `(helm-mode 1)` (e.g. in the init script), which I actually do.

For these cases, @thierryvolpiatto suggested:

> To go out of the loop when you have finished choosing buffers
> hit C-RET (See your mode line) to exit with an empty prompt, you will
> access then to a prompt to occur itself.

but if I do that (which works) I lose the ability to see the results of the search as I type, which is a bummer. The original thread is a bit old, so I was wondering if this is the way one is supposed to be using `multi-occur`, or if I am missing something.

In other words, is there any way to use `multi-occur` in an interactive way while seeing the results as one types the regex string? (assuming that one uses `helm-mode`)
